pkg/clusteragent/autoscaling/workload/pod_patcher.go

```diff
@@ -132,3 +166,37 @@ func (pa podPatcher) findAutoscaler(pod *corev1.Pod) (*model.PodAutoscalerIntern
 
 	return &podAutoscalers[0], nil
 }
+
+func (pa podPatcher) shouldObservePod(pod *workloadmeta.KubernetesPod) bool {
+	return pod.Annotations[model.RecommendationIDAnnotation] != "" &&
+		pod.Annotations[model.AutoscalerIDAnnotation] != "" &&
+		pod.Annotations[model.RecommendationAppliedEventGeneratedAnnotation] == ""
+}
+
+func (pa podPatcher) observedPodCallback(ctx context.Context, pod *workloadmeta.KubernetesPod) {
+	if !pa.isLeader() {
+		return
+	}
+
+	podObj := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pod.Name,
+			Namespace: pod.Namespace,
+			UID:       types.UID(pod.ID),
+		},
+	}
+
+	pa.eventRecorder.AnnotatedEventf(podObj,
```

Comment:
I think that on DCA restart, we will emit all the events again.

Reply:
No because we set an annotation on POD when we trigger an event, it's costly but hooking to Pod create event is not really possible without being somewhat ugly ATM.